lib/Froxlor/Dns/Dns.php

```diff
@@ -549,7 +549,7 @@ private static function generateDkimEntries(array $domain): array
 			// start
 			$dkim_txt = 'v=DKIM1;';
 			// key
-			$dkim_txt .= 'k=rsa;p=' . trim($domain['dkim_pubkey']) . ';';
+			$dkim_txt .= 'k=rsa;p=' . trim(str_replace(PHP_EOL, '', $domain['dkim_pubkey'])) . ';';
```

Comment:
that would only fix half of the issue you are referring to, as it would still include `-----BEGIN PUBLIC KEY-----`and `-----END PUBLIC KEY-----` parts which should not even be there, as generated keys are generated via `rspamd` with parameter `-o plain` which only returns the key itself in one line. 

Still guess there was an update issue when you've upgraded from 2.1 to 2.2.x somehow